catalog/ddms/catalog-ddms-transformer/src/main/java/org/codice/alliance/ddms/v2/Ddms20InputTransformer.java

```diff
@@ -0,0 +1,762 @@
+      return new Ddms20XmlReader(xmlStreamReader).read();
+    } catch (XMLStreamException e) {
+      throw new CatalogTransformerException("Failed to parse DDMS 2.0", e);
+    }
+  }
+
+  private Attribute getTitle(DdmsResource ddms) {
+    String title = ddms.getTitles().stream().findFirst().map(Title::getValue).orElse(null);
+    return new AttributeImpl(Core.TITLE, title);
+  }
+
+  private Attribute getLocation(DdmsResource ddms) {
+    GeometryCollectionBuilder geometryCollectionBuilder = new GeometryCollectionBuilder();
+    for (GeospatialCoverage geospatialCoverage : ddms.getGeospatialCoverages()) {
+      for (BoundingBox boundingBox : geospatialCoverage.getBoundingBoxes()) {
+        geometryCollectionBuilder.polygon(
+            boundingBox.getWest(),
+            boundingBox.getNorth(),
+            boundingBox.getEast(),
+            boundingBox.getNorth(),
+            boundingBox.getEast(),
+            boundingBox.getSouth(),
+            boundingBox.getWest(),
+            boundingBox.getSouth(),
+            boundingBox.getWest(),
+            boundingBox.getNorth());
+      }
+
+      for (BoundingGeometry geometry : geospatialCoverage.getBoundingGeometries()) {
+        geometry
+            .getPolygons()
+            .stream()
+            .map(Polygon::toString)
+            .map(this::convertGmlToWkt)
+            .filter(StringUtils::isNotBlank)
+            .forEach(geometryCollectionBuilder::addWkt);
+
+        geometry
+            .getPoints()
+            .stream()
+            .map(Point::toString)
+            .map(this::convertGmlToWkt)
+            .filter(StringUtils::isNotBlank)
+            .forEach(geometryCollectionBuilder::addWkt);
+      }
+    }
+    String location = geometryCollectionBuilder.toString();
+    // can't return blank string or else the solar catalog provider will throw a
+    // NullPointerException
+    if (StringUtils.isBlank(location)) {
+      location = null;
+    }
+
+    return new AttributeImpl(Core.LOCATION, location);
+  }
+
+  private String convertGmlToWkt(String gmlXml) {
+    try {
+      return gml3ToWkt.convert(gmlXml);
+    } catch (ValidationException e) {
+      // TODO: Should we throw a catalog transform exception or log?
+      return "";
+    }
+  }
+
+  private Attribute getResourceUri(DdmsResource ddms) {
+    String resourceUri =
+        ddms.getIdentifiers()
+            .stream()
+            .filter(identifier -> "urn:catalog:resource-uri".equals(identifier.getQualifier()))
+            .map(Identifier::getValue)
+            .findFirst()
+            .orElse(null);
+    return new AttributeImpl(Core.RESOURCE_URI, resourceUri);
+  }
+
+  private Attribute getResourceDownloadUrl(DdmsResource ddms) {
+    String resourceDownloadUrl =
+        ddms.getRelatedResources()
+            .stream()
+            .filter(relatedResources -> "resource".equals(relatedResources.getRelationship()))
```

Comment:
Metacard.RESOURCE

Reply:
This is a ddms attribute, not a metacard attribute